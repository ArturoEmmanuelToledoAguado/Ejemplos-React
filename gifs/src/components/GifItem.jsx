export const GifItem = ({title,url}) => {

    console.log(url)

    const copyPath = ()=>{
        navigator.clipboard.writeText(url)
    }

  return (
    <div className="">
        <img 
            src={url}
            alt={title}
        />
        <p>{title}</p>
        <button className=""onClick={copyPath}>Copy Path</button>
    </div>
  )
}
