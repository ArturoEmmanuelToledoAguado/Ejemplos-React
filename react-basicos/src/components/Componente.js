import React from "react";
//import React,{Component} from "react";

const Componente=props=> <h2>{props.msg}</h2>

/* Componente en funcion declarada
function Componente(props) {
    return <h2>{props.msg}</h2>;
}*/

/* Componente con clase
class Componente extends Component{
    render() {
        return <h2>{this.props.msg}</h2>;
    }
}*/

export default Componente;