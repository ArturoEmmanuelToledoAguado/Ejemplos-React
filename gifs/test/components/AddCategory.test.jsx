import { fireEvent, render, screen } from "@testing-library/react"
import { AddCategory } from "../../src/components/AddCategory"

describe('Pruebas en <AddCategory/>', () => { 

    test('Esta prueba es para hacer match con el snapshot', () => { 
        const {container} = render(<AddCategory onNext={() =>{}}/>)
        expect(container).toMatchSnapshot()
     })//Evalua que este haciendo algo

    test('Esta prueba es para comprobar la escritura en el componente', () => { 
        const inputValue='Evangelion'
        const onNewCategory=jest.fn()//Toma el evento de que se esta escribiendo, emula la funci√≥n
        render(<AddCategory onNext={onNewCategory}/>)
        const input=screen.getByRole('textbox')
        fireEvent.input(input,{target:{value:inputValue}})
        expect(inputValue).toBe(inputValue)
     })

    test('Esta es una prueba para mandar a llamar el onCategory si el input tiene un valor', () => { 
        const onNewCategory=jest.fn()
        render(<AddCategory onNewCategory={onNewCategory}/>)
        const inputValue='Evangelion'
        const input = screen.getByRole('textbox')
        const form = screen.getByRole('form')
        fireEvent.input(input, {target:{value:inputValue}})
        fireEvent.submit(form)
        expect(input.value).toBe('')
        expect(onNewCategory).toHaveBeenCalled()
        expect(onNewCategory).toHaveBeenCalledWith(inputValue)
        expect(onNewCategory).toHaveBeenCalledTimes(1)
     })

    test('Esta prueba no debe de llamar el on New Category si el input esta vacio', () => { 
        const onNewCategory=jest.fn()
        render(<AddCategory onNewCategory={onNewCategory}/>)

        const form=screen.getByRole('form')
        fireEvent.submit(form)
        expect(onNewCategory).not.toHaveBeenCalled()
        expect(onNewCategory).toHaveBeenCalledTimes(0)
     })
 })