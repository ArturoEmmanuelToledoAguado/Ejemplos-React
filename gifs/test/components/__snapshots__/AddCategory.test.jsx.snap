// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AddCategory/> Esta prueba es para hacer match con el snapshot 1`] = `
<div>
  <form
    aria-label="form"
  >
    <input
      placeholder="Write Something"
      type="text"
      value=""
    />
  </form>
</div>
`;
